### ISSUE TITLE
No 'Access-Control-Allow-Origin' header

### ISSUE THREAD

**Issue by tiagobrito** [lehel-twilio/plugin-dialpad#16]
Access to fetch at 'https://****/create-new-task' from origin 'https://flex.twilio.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by jgcarlson** [comment]
Hey Tiago,
If you're using Twilio Functions, make sure that "Check for valid Twilio signature" is NOT ticked.

**Comment by erperejildo** [comment]
hi @jgcarlson same problem here and that check is not ticked:

**Comment by Karim-Yambu** [comment]
@jgcarlson I am also having the same issue, even the "Check for valid Twilio signature" is NOT ticked.

![image](https://user-images.githubusercontent.com/31468765/57439992-98f30100-7260-11e9-8928-db1068057a71.png)


Can you suggest.

**Comment by lehel-twilio** [comment]
@Karim-Yambu Is this still occurring on version 3.2 of the dialpad?

**Comment by NekoLopez** [comment]
Hello, 
I also have the same problem when I want to make a phone call

![001](https://user-images.githubusercontent.com/31256953/59796784-da38f080-92a3-11e9-8a33-70361a7c4b33.png)

My functions are created and "Check for valid Twilio signature" option is unchecked on each one:

![Screen Shot 2019-06-19 at 2 53 50 PM](https://user-images.githubusercontent.com/31256953/59797514-7b747680-92a5-11e9-994e-f909f9d37309.png)

I'd like your help.

**Comment by lehel-twilio** [comment]
Can you please upgrade to version 4.0 and update with the results? I should have some time over the next couple of days to help you out.

**Comment by NekoLopez** [comment]
> Can you please upgrade to version 4.0 and update with the results? I should have some time over the next couple of days to help you out.

In my situation, I could fix it adding CORS headers in Create New Task which is showing the error: https://www.twilio.com/docs/runtime/functions/faq#how-do-i-send-cors-headers

![007](https://user-images.githubusercontent.com/31256953/61082528-44e6d300-a3ef-11e9-85f5-d672acc9f51c.png)

But now It's showing another error in dialpad-plugin.js, I posted it in a new issue: [https://github.com/lehel-twilio/plugin-dialpad/issues/40#issue-466499458](https://github.com/lehel-twilio/plugin-dialpad/issues/40#issue-466499458)

I updated dialpad plugin and flex to its latest version.

**Comment by lehel-twilio** [comment]
Take a look at /src/functions/create-new-task.js

It already includes the CORS headers:

`  let client = context.getTwilioClient();

  const response = new Twilio.Response();
  response.appendHeader('Access-Control-Allow-Origin', '*');
  response.appendHeader('Access-Control-Allow-Methods', 'OPTIONS POST');
  response.appendHeader('Content-Type', 'application/json');
  response.appendHeader('Access-Control-Allow-Headers', 'Content-Type');`

**Comment by NekoLopez** [comment]
> Take a look at /src/functions/create-new-task.js
> 
> It already includes the CORS headers:
> 
> let client = context.getTwilioClient();
> 
> const response = new Twilio.Response();
> response.appendHeader('Access-Control-Allow-Origin', '*');
> response.appendHeader('Access-Control-Allow-Methods', 'OPTIONS POST');
> response.appendHeader('Content-Type', 'application/json');
> response.appendHeader('Access-Control-Allow-Headers', 'Content-Type');

Yes! You were right, I had to create all the functions according to functions folder, but now when I want to make a phone call it's showing an error on Twilio debugger: Got HTTP 500 response from https://*****.twil.io/call-outbound-join Error: 'From' phone number not verified. My Twilio phone number is not from USA. How can I fix it?

**Comment by lehel-twilio** [comment]
Take a look at this:

https://support.twilio.com/hc/en-us/articles/223180048-Adding-a-Verified-Phone-Number-or-Caller-ID-with-Twilio

I would like to close this issue down, as there are no issues with the current code.
